app/Utils/HasPeriodicEvent.php

```diff
@@ -44,6 +44,21 @@ final public function periodicEvent(): ?PeriodicEvent
             ->first();
     }
 
+    /**
+     * Get the PeriodicEvent connected to the controller
+     * and belonging to the given semester
+     * (by default the current one).
+     *
+     * @return PeriodicEvent|null
+     */
+    final public function periodicEventForSemester(?Semester $semester): ?PeriodicEvent
+    {
+        if (is_null($semester)) $semester = Semester::current();
+        return PeriodicEvent::where('event_model', $this->underlyingControllerName)
+            ->where('semester_id', $semester->id)
+            ->first();
+    }
+
```

Comment:
@kdmnk that's what I thought of. And we call this method with the semester for which we are creating the question. And if that semester is in the future, this will return null, and the question will not be opened.

Reply:
This won't give you any data for older semesters. 
The information flow should be: you get/set the periodicEvent, and through that, you get the semester that you can use. 

When you create a questions, you store them with the semester you get from the current periodicEvent.
Afterwards, you can still query older questions by directly using the semester ids, without the periodicEvent.

Reply:
Oh; so those are only there for querying older ones... Maybe the easiest way for now would be to only allow admins to add questions for the current semester. (They probably won't want to do so for future semesters anyway.)

Let me know what you think.

Reply:
Well if you need this asap I think go back to the embedded google forms :D 

We could try to release it, but I'm trying not to release half done features anymore. I will have time during the weekend to have a deeper look again and give some solutions for our uncertainties. (Currently the morph relation's complexity overcomes its advantages, but I have to take another look).

Again, after the previous years of experience, our priority here is to produce software that is easily maintainable and understandable, which is more important to me than releasing new features.
After the discussions, I might also need to go back to the PeriodicEvents as I can see it is not clearly understandable and does not really cover this case. 

What do you think?

Reply:
I also understand if you want to release this and will go back to it afterwards if you are confident about the current solution's behaviour.

Reply:
> I also understand if you want to release this and will go back to it afterwards if you are confident about the current solution's behaviour.

Yes, that would be the best for me now:) and after the form gets closed, we can open a new issue for it. I think I can work on this more on July.

Thank you 😊

Reply:
I advise against deploying changes in the codebase that are not sure to be correct. I also have a problem on an operational standpoint as in June there will not be enough human resources to restore a backup if that is needed due to a mistake presented in the newly deployed code. But I will deploy the changes if someone takes the responsibility even though I would not recommend it.

Reply:
Okay; I think I've played with it long enough to say it's going to be alright. So I'll take the blame if something happens.